### PR TITLE
Email integration and secret/env setup

### PR DESCRIPTION
**Pull request by Ruben-Brits**
Please make sure that everyone who wants the emails to work locally has to add their own **secrets.properties** file as specified in the template. I will send the api key personally when needed to keep it safe.

### CODE REVIEW

File: Server/pom.xml [hunk 1]
```diff
@@ -48,11 +48,22 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-configuration-processor</artifactId>
+            <optional>true</optional>
+        </dependency>
 		<dependency>
 			<groupId>org.bouncycastle</groupId>
 			<artifactId>bcpkix-jdk15on</artifactId>
 			<version>1.65</version>
 		</dependency>
+		<!-- https://mvnrepository.com/artifact/com.sendgrid/sendgrid-java -->
```
Comment (ricecakesreece) [inline Server/pom.xml:61]: is this meant to be commented out?

File: Server/src/main/java/Database/TheAuPair/Controllers/EmailController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          request.setEndpoint("mail/send");
+          request.setBody(mail.build());
+          response = sendGridClient.api(request);
+        } catch (IOException ex) {
+          System.out.println(ex.getMessage());
+        }
+    
+        return response;
+      }
+}
```
Comment (ricecakesreece) [inline Server/src/main/java/Database/TheAuPair/Controllers/EmailController.java:60]: no new line

File: Server/src/main/resources/application.properties [hunk 3]
```diff
@@ -1,2 +1,4 @@
+ spring.config.import=optional:secrets.properties
+
 spring.data.mongodb.uri=mongodb+srv://Cheems:e7nM0XHs8KcpXtdO@cluster0.oj3gk.mongodb.net/The-Au-Pair?retryWrites=true&w=majority
-spring.data.mongodb.database=The-Au-Pair
+spring.data.mongodb.database=The-Au-Pair
```
Comment (ricecakesreece) [inline Server/src/main/resources/application.properties:4]: newline

File: Server/src/main/resources/secrets_template.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+sendgrid.api-key=OUR_SENDGRID_API_KEY
```
Comment (ricecakesreece) [inline Server/src/main/resources/secrets_template.properties:1]: newline

**Review by ricecakesreece — APPROVED** [review]
Amazing

**Review by priyo5 — APPROVED** [review]

**Pull request merged**